onadata/libs/serializers/data_serializer.py

```diff
@@ -28,10 +28,15 @@ def get_request_and_username(context):
     request = context['request']
     view = context['view']
     username = view.kwargs.get('username')
+    form_uuid = view.kwargs.get('form_uuid')
 
     if not username:
-        # get the username from the user if not set
-        username = (request.user and request.user.username)
+        # get the username from the XForm object if form_uuid is
+        # present else utilize the request users username
+        if form_uuid:
+            username = XForm.objects.get(uuid=form_uuid).user.username
```

Comment:
This state will most likely happen every time Enketo makes a submission. Other than that the only other time this would happen is if a client decided to submit date with the new endpoint.

Reply:
Can you please verify that `XForm` has an index on the `uuid` column

Reply:
> Can you please verify that `XForm` has an index on the `uuid` column

The `uuid` column is currently not indexed

Reply:
added index migration looks good